suite-native/module-device-settings/src/screens/AutoConnectSettingsScreen.tsx

```diff
@@ -0,0 +1,118 @@
+import { useThpAutoconnectActions } from '@suite-native/thp';
+import { useToast } from '@suite-native/toasts';
+import TrezorConnect from '@trezor/connect';
+
+import { useDeviceConnectionGuard } from '../hooks/useDeviceConnectionGuard';
+
+type NavigationProp = StackNavigationProps<
+    DeviceSettingsStackParamList,
+    DeviceSettingsStackRoutes.DeviceSettings
+>;
+
+export const AutoConnectSettingsScreen = () => {
+    const { isDeviceConnected } = useDeviceConnectionGuard();
+
+    const dispatch = useDispatch();
+    const navigation = useNavigation<NavigationProp>();
+
+    const { translate } = useTranslate();
+    const { showToast } = useToast();
+    const { startThpAutoconnect } = useThpAutoconnectActions();
+
+    const autoconnectCredentials = useSelector(selectDeviceAutoconnectCredentials);
+
+    const isAutoconnectEnabled = autoconnectCredentials.length > 0;
+
+    const onAutoconnectTurnOff = useCallback(async () => {
+        const result = await dispatch(
+            removeThpAutoconnectThunk({
+                credentials: autoconnectCredentials,
+            }),
+        ).unwrap();
+
+        if (isRejected(result)) {
+            showToast({
+                variant: 'error',
+                message: result.error.message,
+            });
+        }
+    }, [dispatch, showToast, autoconnectCredentials]);
+
+    const onAutoconnectTurnOn = useCallback(async () => {
+        navigation.navigate(DeviceSettingsStackRoutes.ContinueOnTrezor);
+
+        const result = await startThpAutoconnect();
+
+        if (result && isRejected(result)) {
+            TrezorConnect.cancel();
+            showToast({
+                variant: 'error',
+                message: <Translation id="moduleDeviceSettings.autoconnect.enable.error" />,
+            });
+        }
+        navigation.goBack();
+    }, [navigation, startThpAutoconnect, showToast]);
+
+    const onPress = () => {
+        if (isAutoconnectEnabled) {
+            onAutoconnectTurnOff();
+        } else {
+            onAutoconnectTurnOn();
+        }
+    };
+
+    if (!isDeviceConnected) return null;
+
+    return (
+        <Screen
+            header={
+                <DynamicScreenHeader
+                    title={<Translation id="moduleDeviceSettings.autoconnect.settingsCard.title" />}
+                    subtitle={<Translation id="moduleDeviceSettings.autoconnect.screen.subtitle" />}
+                    closeActionType="close"
+                />
+            }
+        >
+            <Card>
+                <VStack spacing="sp32">
+                    <PictogramTitleHeader
+                        variant="success"
+                        icon={isAutoconnectEnabled ? 'check' : 'x'}
+                        title={
+                            isAutoconnectEnabled ? 'Auto connect enabled' : 'Auto connect disabled'
+                        }
+                        subtitle={
+                            isAutoconnectEnabled
```

Comment:
Might make sense to have an object with all the translations and configuration options as we use elsewhere, e.g. in `BluetoothDeviceList`.

Reply:
I'll do this when final copy is available.